db/external_sst_file_test.cc

```diff
@@ -3719,17 +3719,455 @@ TEST_F(ExternalSSTFileWithTimestampTest, TimestampsNotPersistedBasic) {
+    ASSERT_EQ(1, live_meta.size());
+    ASSERT_EQ(live_meta[0].largest_seqno, 0);
+
+    ASSERT_OK(Flush(1));
+    ASSERT_OK(db_->IngestExternalFile(
+        handles_[1],
+        {live_meta[0].directory + "/" + live_meta[0].relative_filename},
+        ingest_opts));
+    // Will write DB state to a new MANIFEST during open.
+    ASSERT_OK(TryReopenWithColumnFamilies({"default", "toto"}, options));
+    // read from new MANIFEST
+    ASSERT_OK(TryReopenWithColumnFamilies({"default", "toto"}, options));
+
+    // If global seqno is not used for the ingested file, keys in it have seqno
+    // 0 and will be dropped during compaction.
+    ASSERT_OK(db_->CompactRange(cro, handles_[1], nullptr, nullptr));
+    std::string val;
+    for (int k = 0; k < kNumKeys; ++k) {
+      ASSERT_OK(db_->Get(ReadOptions(), handles_[1], Key(k), &val));
+      ASSERT_EQ(val, "val_" + Key(k));
+    }
+  } while (ChangeOptions(kSkipPlainTable | kSkipFIFOCompaction));
+}
+
+TEST_P(IngestLiveDBFileTest, Basic) {
+  if (encrypted_env_ && ingest_opts.move_files) {
+    ROCKSDB_GTEST_SKIP(
+        "Encrypted env and move_files do not work together as we reopen the "
+        "file after linking it, which would append an extra encryption "
+        "prefix.");
+    return;
+  }
+  do {
+    SCOPED_TRACE("option_config_ = " + std::to_string(option_config_));
+    Options options = CurrentOptions();
+    BlockBasedTableOptions table_options;
+    table_options.index_type =
+        BlockBasedTableOptions::kBinarySearchWithFirstKey;
+    options.table_factory.reset(NewBlockBasedTableFactory(table_options));
+    CreateAndReopenWithCF({"toto"}, options);
+
+    // Generate a file to ingest in default CF
+    for (int k = 0; k < kNumKeys; ++k) {
+      ASSERT_OK(Put(Key(k), "val_" + std::to_string(k)));
+      ASSERT_OK(Put(1, Key(k), "base_" + std::to_string(k)));
+    }
+    const Snapshot* snapshot = db_->GetSnapshot();
+    ASSERT_OK(Flush());
+
+    // Case 1. Ingest a file with non-zero seqno and a file boundary key has
+    // non-zero seqno, this should fail.
+    //
+    // To get the largest internal key of a file, need to get FileMetaData.
+    std::vector<std::vector<FileMetaData>> metadata;
+    dbfull()->TEST_GetFilesMetaData(db_->DefaultColumnFamily(), &metadata,
+                                    nullptr);
+    const FileMetaData& file = metadata[0][0];
+    ValueType vtype;
+    SequenceNumber seq;
+    UnPackSequenceAndType(ExtractInternalKeyFooter(file.largest.Encode()), &seq,
+                          &vtype);
+    ASSERT_GE(seq, 0);
+
+    std::vector<LiveFileMetaData> live_meta;
+    db_->GetLiveFilesMetaData(&live_meta);
+    ASSERT_EQ(1, live_meta.size());
+    ASSERT_GT(live_meta[0].largest_seqno, 0);
+    std::string ingest_file_path =
+        live_meta[0].directory + "/" + live_meta[0].relative_filename;
+    Status s =
+        db_->IngestExternalFile(handles_[1], {ingest_file_path}, ingest_opts);
+    ASSERT_NOK(s);
```

Comment:
Where is the non-OK Status for this returned, I didn't find this change in the PR, I could be missing it?